internal/devbox/generate/tmpl/prod.Dockerfile.tmpl

```diff
@@ -0,0 +1,28 @@
+FROM jetpackio/devbox:latest
+
+WORKDIR /code
+USER root:root
+RUN mkdir -p /code && chown ${DEVBOX_USER}:${DEVBOX_USER} /code
+USER ${DEVBOX_USER}:${DEVBOX_USER}
+
+{{- /*
+Ideally, we first copy over devbox.json and devbox.lock and run `devbox install` 
+to create a cache layer for the dependencies. This is complicated because
+devbox.json may include local dependencies (flakes and plugins). We could try
+to copy those in (the way the dev Dockerfile does) but that's brittle because
+those dependencies may also pull in other local dependencies and so on. Another
+sulution would be to add a new flag `devbox install --skip-errors` that would 
+just try to install what it can, and ignore the rest.
+
+A hack to make this simpler is to install from the lockfile instead of the json.
+*/}}
+
+COPY --chown=${DEVBOX_USER}:${DEVBOX_USER} . .
+
+RUN devbox install
+
+RUN {{ .DevboxRunInstall }}
+
+RUN {{ .DevboxRunBuild }}
+
+CMD [{{ .Cmd }}]
```

Comment:
It think we can optimize this image a bit. Something like:
- gc nix store (clean up)
- devbox shellenv (so cmd has the right env)
- separate install layer

Reply:
@LucilleH for sure! I think some of those might be doable without devbox changes (but not all).

> gc nix store (clean up)

I can try this, but in theory the nix store should be pretty clean no? We haven't installed anything we don't need.

> devbox shellenv

This can help a bit but not without changes. Ideally, we do shellenv and then call the raw start command (without using `devbox run`). A small concern is the init hook not running multiple times (I think that's probably fine). So in practice it would look like:

```
RUN eval $(devbox shellenv --init-hook)
RUN devbox run install --raw
RUN devbox run build --raw
CMD ["devbox", "run", "start", "--raw"] 
```

Where `--raw` would be a new flag that indicates run the raw script without starting a devbox env (which would be wasteful) 

> separate install layer

This is the biggest win. But see my comment about local dependencies. I think there's 2 huge optimizations here:

* Do `devbox install` before copying source code. This makes a very cachable base layer.
* Split stages so that we copy over only nix stuff we need into final image. 

(1) is easy, but requires a new flag/command so we can ignore errors of missing local dependencies.
(2) is harder, but we did something similar in early devbox days.

My preference is to merge this and optimize in follow up. Thoughts?